File: GMS-frontend/src/app/endpoints/clanarine-endpoints/clanarine-edit-endpoint.ts

import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {Config} from "../../config";
import {Observable} from "rxjs";
@Injectable({providedIn: 'root'})
export class ClanarineEditEndpoint implements  MyBaseEndpoint<ClanarineEditRequest, number>{
  constructor(public httpClient:HttpClient) { }

  Handle(request: ClanarineEditRequest): Observable<number> {
    let url=Config.adresa + `Clanarina-Edit`;

    return this.httpClient.post<number>(url, request);
  }

}
export interface ClanarineEditRequest {
  id: number
  naziv: string
  cijena: number
  opis: string
}

File: GMS-backend/GMS/Entities/Endpoint/Clanarina/Edit/ClanarineEditEndpoint.cs
using GMS.Data;
using GMS.Helpers;
using GMS.Helpers.Auth;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace GMS.Entities.Endpoint.Clanarina.Edit
{

    [Route("Clanarina-Edit")]
    [MyAuthorization]

    public class ClanarineEditEndpoint : MyBaseEndpoint<ClanarineEditRequest, int>
    {
        private readonly ApplicationDbContext db;

        public ClanarineEditEndpoint(ApplicationDbContext db)
        {
            this.db = db;
        }

        [HttpPost]
        public override async Task<int> Handle([FromBody] ClanarineEditRequest request, CancellationToken cancellationToken)
        {
            Models.Clanarina? clanarina;
            if (request.ID == 0)
            {
                clanarina = new Models.Clanarina();
                db.Add(clanarina);


            }
            else
            {
                clanarina = db.Clanarina.FirstOrDefault(s => s.ID == request.ID);
                if (clanarina == null)
                    throw new Exception("pogresan ID");
            }

            clanarina.Naziv = request.Naziv.RemoveTags();
            clanarina.Cijena = request.Cijena;
            clanarina.Opis = request.Opis.RemoveTags();



            await db.SaveChangesAsync(cancellationToken);

            return clanarina.ID;
        }
    }
}

